From f4256533d1ffdc203c3f8c6ee42e8dcde470a93f Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Tue, 11 Jan 2022 21:06:35 -0500
Subject: [PATCH] use more idiomatic/correct embedded shell scripting

Formatting arguments into code is fragile, and meson natively knows how
to pass files as arguments via argv. So pass them as arguments to the
shell.
---
 doc/dbus/meson.build | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/doc/dbus/meson.build b/doc/dbus/meson.build
index 802938e..861da64 100644
--- a/doc/dbus/meson.build
+++ b/doc/dbus/meson.build
@@ -19,11 +19,10 @@ foreach interface_file: dbus_interfaces_files
         capture: true,
         command: [
             bash, '-c',
-            'cat @INPUT@;' +
-            xsltproc.path() + ' @0@/@1@ '.format(
-                meson.source_root(),
-                files('spec-to-docbook.xsl')[0]) +
-            interface_file.full_path() + '| tail -n +2;',
+            'cat "$1"; "$2" "$3" "$4" | tail -n +2',
+            '_', # argv0 ignored
+            '@INPUT@',
+            xsltproc, files('spec-to-docbook.xsl')[0], interface_file
         ],
     )
 endforeach
-- 
GitLab

